### DISCUSSION TITLE
Cannot open the connection error

### DISCUSSION THREAD

**Discussion by psychobas** [ropensci/targets#995] [Help]
I run a large number of targets in parallel with `tar_make_clustermq`, with 23 workers. Each target usually takes several hours or even days to finish. After running for usually about two to three days, I get the following error:

<img width="348" alt="image" src="https://user-images.githubusercontent.com/44177662/206843715-5df86ebb-cf2c-4879-8d3f-6d7059da9e12.png">

This message is printed about 100 times before the workers finally shut down with the following final error message:

<img width="702" alt="image" src="https://user-images.githubusercontent.com/44177662/206843882-29766365-5a6c-4d97-a334-13905e68936b.png">

The final error mentions the rlang package, but I have this package installed and listed in the packages in `tar_option_set`. I run `tar_make_clustermq` as a job in Rstudio on a Linux VM. I deactivated R Studio autosave and file indexing, both didn't solve the problem. Maybe it occurs when two workers finish a target at the same time and thus try to open _targets/meta/meta at the same time?

I have targets version 0.11.0 installed. I would appreciate any help!

**Comment by wlandau** [comment]
First I recommend debugging on a scaled-down version of the pipeline that runs in a couple seconds (rather than a couple days) but still gives the same set of errors. Also would recommend looking for clues by trying `tar_make(callr_function = NULL)` and `tar_make_clustermq(callr_function = NULL)`. (More on this at https://books.ropensci.org/targets/debugging.html#system-issues and elsewhere in that chapter.) 

`targets` works by starting a `callr`-powered controller process that reads `_targets.R` and launches the workers. Only the controller reads and writes to `_targets/meta/meta` (not the workers) it looks like the controller somehow has a different set of permissions or even runs as a different user than your interactive R session. It may be worth comparing your user and permissions in the interactive session vs a process invoked by `callr::r()`.

The workers do not write to `_targets/meta/meta`, only the controller process does. So it is most likely not a race condition. However, strange issues are known to happen if there is severe lag communicating with the file system that has `_targets/meta/meta`. This can happen if the data store is on a mounted network drive connected through a slow network.
